### PR TITLE
Update pyright, pandas, mypy versions.  Fix nightly tests

### PR DESCRIPTION
**Pull request by Dr-Irv**
Fixed the nightly tests.  Revealed some bugs in the dev code.

### CODE REVIEW

File: tests/test_groupby.py [hunk 1]
```diff
@@ -246,15 +246,49 @@ def df2scalar(val: DataFrame) -> float:
...
-        check(assert_type(GB_DF.resample("ME").interpolate(), DataFrame), DataFrame)
-        check(
-            assert_type(GB_DF.resample("ME").interpolate(method="linear"), DataFrame),
-            DataFrame,
-        )
-        check(
-            assert_type(GB_DF.resample("ME").interpolate(inplace=True), None),
-            type(None),
-        )
+        try:
```
Comment (twoertwein) [inline tests/test_groupby.py:249]: should this be `if PD_LTE_22:`?

**Review by twoertwein — COMMENTED** [review]
Looks good to me, but I don't understand what's the difference between `try` and ` if PD_LTE_22`

File: tests/test_groupby.py (hunk 1 above)
Comment (Dr-Irv) [reply tests/test_groupby.py]: Good point.  Not sure why I went that direction.  Fixed in https://github.com/pandas-dev/pandas-stubs/pull/920/commits/47aac27bde61fb6cd188b770c84f773eb529092b

**Review by twoertwein — APPROVED** [review]

**Comment by twoertwein** [comment]
Thank you @Dr-Irv !

**Comment by twoertwein** [comment]
The CI failed on main:

> tests/test_api_types.py:412: error: Expression is of type "Any", not "DataFrame"  [assert-type]
> tests/test_api_types.py:412: error: Module has no attribute "from_dataframe"  [attr-defined]

**Comment by Dr-Irv** [comment]
> The CI failed on main:
> 
> > tests/test_api_types.py:412: error: Expression is of type "Any", not "DataFrame"  [assert-type]
> > tests/test_api_types.py:412: error: Module has no attribute "from_dataframe"  [attr-defined]

Weird `mypy` behavior.  Was working locally for me with version 1.10.  Updated poetry environment, which only updated as follows:
```text
  - Updating setuptools (68.2.2 -> 69.5.1)
  - Updating lxml (5.2.1 -> 5.2.2)
  - Updating xarray (2024.3.0 -> 2024.5.0)
```
then it failed locally.  Created a fix in #922

**Pull request merged**